src/server/pfs/server/driver_new_storage.go

```diff
@@ -298,24 +299,105 @@ func (fr *FileReader) drain() error {
 	return nil
 }
 
-func (d *driver) compact(master *work.Master, outputPath string, inputPrefixes []string) (retErr error) {
+// compact is the entrypoint to compaction.
+func (d *driver) compact(ctx context.Context, master *work.Master, outputPath string, inputPrefixes []string) error {
+	// resolve prefixes into paths
+	inputPaths := []string{}
+	for _, inputPrefix := range inputPrefixes {
+		err := d.storage.WalkFileSets(ctx, inputPrefix, func(inputPath string) error {
+			inputPaths = append(inputPaths, inputPath)
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+	}
+	if d.env.StorageCompactionMaxFanIn < 2 {
```

Comment:
We probably should not get to this point if this is the case, we probably want this grouped with other validation (I think we can punt on this tbh).

Reply:
I guess I can remove it. It will probably OOM pretty quickly anways.

Reply:
Yeah, I think it is worth making a todo (and you could even have the todo somewhere around here), but I think this type of validation should be handled at setup, not while we are in the middle of a compaction.